node/src/components/storage/tests.rs

```diff
@@ -986,73 +1036,98 @@ fn test_legacy_interface() {
+            put_block(&mut harness, &mut storage, block.clone());
+            let mut execution_results = HashMap::new();
+            execution_results.insert(*deploy.id(), execution_result.clone());
+            put_execution_results(&mut harness, &mut storage, *block.hash(), execution_results);
+            assert_eq!(
+                get_block_at_height(&mut storage, 42).expect("block not indexed properly"),
+                *block
+            );
 
-    let actual_block = get_block(&mut harness, &mut storage, *block.hash())
-        .expect("missing block we stored earlier");
-    assert_eq!(actual_block, *block);
+            // After storing everything, destroy the harness and component, then rebuild using the
+            // same directory as backing.
+            let (on_disk, rng) = harness.into_parts();
+            let mut harness = ComponentHarness::builder()
+                .on_disk(on_disk)
+                .rng(rng)
+                .build();
+            let mut storage = storage_fixture(&harness, merkle_tree_hash_activation);
 
-    let actual_deploys = get_deploys(&mut harness, &mut storage, smallvec![*deploy.id()]);
-    assert_eq!(actual_deploys, vec![Some(deploy.clone())]);
+            let actual_block = get_block(&mut harness, &mut storage, *block.hash())
+                .expect("missing block we stored earlier");
+            assert_eq!(actual_block, *block);
 
-    let (_, deploy_metadata) = get_deploy_and_metadata(&mut harness, &mut storage, *deploy.id())
-        .expect("missing deploy we stored earlier");
+            let actual_deploys = get_deploys(&mut harness, &mut storage, smallvec![*deploy.id()]);
+            assert_eq!(actual_deploys, vec![Some(deploy.clone())]);
 
-    let execution_results = deploy_metadata.execution_results;
-    assert_eq!(execution_results.len(), 1);
-    assert_eq!(execution_results[block.hash()], execution_result);
+            let (_, deploy_metadata) =
+                get_deploy_and_metadata(&mut harness, &mut storage, *deploy.id())
+                    .expect("missing deploy we stored earlier");
 
-    assert_eq!(
-        get_block_at_height(&mut storage, 42).expect("block index was not restored"),
-        *block
-    );
+            let execution_results = deploy_metadata.execution_results;
+            assert_eq!(execution_results.len(), 1);
+            assert_eq!(execution_results[block.hash()], execution_result);
+
+            assert_eq!(
+                get_block_at_height(&mut storage, 42).expect("block index was not restored"),
+                *block
+            );
+        });
+    }
 }
 
 #[test]
 fn should_hard_reset() {
     let blocks_count = 8_usize;
     let blocks_per_era = 3;
     let mut harness = ComponentHarness::default();
-    let mut storage = storage_fixture(&harness);
+
+    // `merkle_tree_hash_activation` can be chosen arbitrarily
+    let merkle_tree_hash_activation = harness.rng.gen();
```

Comment:
This test will create blocks in three eras, but `merkle_tree_hash_activation` will almost never be that small. Maybe use a random number from the range `0..4` instead?

Reply:
Yes, that's right, to some extent :)
Because `merkle_tree_hash_activation` is `u64` I'd still like this to be tested against huge values. I think I'll use the `LogNormal` distribution here with parameters strongly preferring the low numbers, but occasionally hitting the big ones.

Reply:
Applied in https://github.com/casper-network/casper-node/pull/2531/commits/5bc3333778683c431c4666489acf3b2e21b12dcc